tests/conftest.py

```diff
@@ -346,21 +351,31 @@ def k8scluster(k8smasters):
     return k8s_master_cluster
 
 @pytest.fixture(scope="module")
-def nbrhosts(ansible_adhoc, tbinfo, creds):
+def nbrhosts(ansible_adhoc, tbinfo, creds, request):
     """
     Shortcut fixture for getting VM host
     """
 
     vm_base = int(tbinfo['vm_base'][2:])
+    neighbor_type = request.config.getoption("--neighbor_type")
```

Comment:
Is there a way to detect the type of neighbor? Then the "--neighbor_type" option would be unnecessary. It would be easier to trigger tests.

Reply:
I don't have any elegant idea to check the type of neighbor.
In my mind, we can only get the type after we login the device, but the password and username is different between EOS and SONiC, so it requires that we need to know the type before we login the device.